backend/test/authorizationTest.py

```diff
@@ -0,0 +1,143 @@
+    """
+    global failed
+    time.sleep(0.1)
+    if not response.status_code == 401:
+        failed += 1
+        logging.debug("Got response: {}".format(response.status_code))
+        logging.debug(response.content)
+        logging.error("Endpoint {0} is not secure ❌".format(endpoint))
+    else:
+        logging.info("Endpoint {0} is secure ✅".format(endpoint))
+
+def test_get(endpoint: str, token: str, params={}):
+    """Test an endpoint that uses a get REST request
+
+    Parameters:
+    endpoint (str): endpoint to test
+    token (str): auth token (JWT) for headers
+    params (dict): parameters for get request
+    """
+    global tests
+    tests += 1
+    headers = get_header(token)
+    url = '{0}{1}'.format(base_url, endpoint)
+    response = requests.get(url, params=params, headers=headers)
+    check_if_unauthorized(response, endpoint)
+
+def test_delete(endpoint: str, token: str):
+    """Test an endpoint that uses a delete REST request
+
+    Parameters:
+    endpoint (str): endpoint to test
+    token (str): auth token (JWT) for headers
+    """
+    global tests
+    tests += 1
+    headers = get_header(token)
+    url = '{0}{1}'.format(base_url, endpoint)
+    response = requests.delete(url, headers=headers)
+    check_if_unauthorized(response, endpoint)
+
+def test_post(endpoint: str, token: str, body={}):
+    """Test an endpoint that uses a post REST request
+
+    Parameters:
+    endpoint (str): endpoint to test
+    token (str): auth token (JWT) for headers
+    body (dict): NOT USED - body for the post request
+    """
+    global tests
+    tests += 1
+    headers = get_header(token)
+    url = '{0}{1}'.format(base_url, endpoint)
+    data = {}
+    # data['body'] = json.dumps(body)
+    response = requests.post(url, data=data, headers=headers)
+    check_if_unauthorized(response, endpoint)
+
+def test_put(endpoint: str, token: str, body={}):
+    """Test an endpoint that uses a put REST request
+
+    Parameters:
+    endpoint (str): endpoint to test
+    token (str): auth token (JWT) for headers
+    body (dict): NOT USED - body for the post request
+    """
+    global tests
+    tests += 1
+    headers = get_header(token)
+    url = '{0}{1}'.format(base_url, endpoint)
+    data = {}
+    # data['body'] = json.dumps(body)
+    response = requests.put(url, data=data, headers=headers)
+    check_if_unauthorized(response, endpoint)
+
+######################### Test config #########################
+
+parser = argparse.ArgumentParser(description="Test security and authorization checks for Daily Dash backend.")
+parser.add_argument('--debug', action='store_true', default=False)
+args = parser.parse_args()
+
+logging.basicConfig()
+if args.debug:
+    logging.getLogger().setLevel(logging.DEBUG)
+else:
+    logging.getLogger().setLevel(logging.INFO)
+
+######################### Run tests #########################
+# Note: Body has no effect, I could not get it working
```

Comment:
I know this might not be necessary for testing authorization, but if you switch from data=data to json=data at line 89, the body in the request should work.